File: Data Structures & Algorithms/longest-substring-without-duplicates/submission-1.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        
        // int arr[26]={0};
        unordered_map<char,int>arr;
        int st =0;
        int curr = 0;
        int ans = 0;
        while(curr<s.length()){
            while(curr<s.length() and (arr.find(s[curr])==arr.end() or arr[s[curr]]==0)){
                arr[s[curr]] = 1;
                curr++;
            }
            ans = max(ans,curr-st);

            while(curr<s.length() and st<curr and arr[s[curr]]!=0){
                arr[s[st]] = 0;
                st++;
            }
        }

        return ans;
    }
};

File: Data Structures & Algorithms/partition-labels/submission-1.py
class Solution:
    def partitionLabels(self, s: str) -> List[int]:
        
        dictt = {}
        for i in range(len(s)):
            if s[i] in dictt:
                dictt[s[i]][1] = i
            else:
                dictt[s[i]] = [i,i]
        print(dictt)
        ans = []
        maxx = -1
        curr = 0
        for i in range(len(s)):
            maxx = max(maxx,dictt[s[i]][1])
            curr = curr + 1
            if i == maxx:
                ans.append(curr)
                curr = 0
  
        return ans